samples/QmlArExample/main.cpp

```diff
@@ -0,0 +1,171 @@
+// without modification, provided you include the original copyright
+// notice and use restrictions.
+//
+// See the Sample code usage restrictions document for further information.
+//
+
+#include <QDebug>
+#include <QSettings>
+#include <QGuiApplication>
+#include <QQmlApplicationEngine>
+#include <QQuickWindow>
+#include <QCommandLineParser>
+#include <QDir>
+#include <QSurfaceFormat>
+
+#ifdef Q_OS_WIN
+#include <Windows.h>
+#endif
+
+#include "AppInfo.h"
+#include "QmlArcGISArView.h"
+#include "LocationDataSource.h"
+
+//------------------------------------------------------------------------------
+
+#define kSettingsFormat                 QSettings::IniFormat
+
+//------------------------------------------------------------------------------
+
+#define kArgShowName                    "show"
+#define kArgShowValueName               "showOption"
+#define kArgShowDescription             "Show option maximized | minimized | fullscreen | normal | default"
+#define kArgShowDefault                 "show"
+
+#define kShowMaximized                  "maximized"
+#define kShowMinimized                  "minimized"
+#define kShowFullScreen                 "fullscreen"
+#define kShowNormal                     "normal"
+#define STRINGIZE(x) #x
+#define QUOTE(x) STRINGIZE(x)
+
+//------------------------------------------------------------------------------
+
+int main(int argc, char *argv[])
+{
+  // There are some conflicts between the AR frameworks and the Qt's rendering thread.
+  // This lines of code enable the non-threaded render loops mode in Qt.
+  // See SceneView::renderFrame documentation and Qt's documentation
+  // https://doc.qt.io/qt-5/qtquick-visualcanvas-scenegraph.html#non-threaded-render-loops-basic-and-windows
+  // for more informati ons.
+#if defined(Q_OS_IOS) || defined(Q_OS_ANDROID)
+  qputenv("QSG_RENDER_LOOP", "basic");
+#endif
+
+#if defined(Q_OS_LINUX) && !defined(Q_OS_ANDROID)
+  // Linux requires 3.2 OpenGL Context
+  // in order to instance 3D symbols
+  QSurfaceFormat fmt = QSurfaceFormat::defaultFormat();
+  fmt.setVersion(3, 2);
+  QSurfaceFormat::setDefaultFormat(fmt);
+#endif
+
+  QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+  QGuiApplication app(argc, argv);
+
+  QCoreApplication::setApplicationName(kApplicationName);
+  QCoreApplication::setApplicationVersion(kApplicationVersion);
+  QCoreApplication::setOrganizationName(kOrganizationName);
+#ifdef Q_OS_MAC
+  QCoreApplication::setOrganizationDomain(kOrganizationName);
+#else
+  QCoreApplication::setOrganizationDomain(kOrganizationDomain);
+#endif
+  QSettings::setDefaultFormat(kSettingsFormat);
+
+  // Initialize license
+
+#ifdef kLicense
+  QCoreApplication::instance()->setProperty("Esri.ArcGISRuntime.license", kLicense);
+#endif
+
+  // Register the QmlArcGISArView type
+  qmlRegisterType<Esri::ArcGISRuntime::Toolkit::QmlArcGISArView>("Esri.ArcGISArToolkit", 1, 0, "ArcGISArViewInternal");
```

Comment:
why is this registering `ArcGISArViewInternal`? Is this correct?

Reply:
It's not correct. I just ran into this problem on iOS when running this example app. I think it needs to be as follows:

```suggestion
  qmlRegisterType<Esri::ArcGISRuntime::Toolkit::QmlArcGISArView>("Esri.ArcGISArToolkit", 1, 0, "ArcGISArView");
```

Reply:
There is a little confusion here.

There is an C++ class called `QmlArcGISArView`. This class is registred to QML engine with the name `ArcGISArViewInternal`. And the QML component `ArcGISArView` is implemented in the file `ArcGISArView.qml` and use`ArcGISArViewInternal` internally. The users can't use directly the C++ class.

Maybe the names are not the best one? Maybe `QmlArcGISArView` should be renamed to `ArcGISArViewInternal`?

Reply:
Ideally users shouldn't have to register any class with internal in the name. Is that possible?

Reply:
I added `QmlArcGISArView::qmlRegisterTypes()` to do that. In the future, If we provide the binaries of the AR toolkit, this can be replaced by `qmldir` and `.qmlfiles` in a QML plugin.